src/test/java/org/junitpioneer/jupiter/resource/ResourcesTests.java

```diff
@@ -0,0 +1,543 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.resource;
+
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.junit.platform.testkit.engine.EventConditions.finished;
+import static org.junit.platform.testkit.engine.TestExecutionResultConditions.cause;
+import static org.junit.platform.testkit.engine.TestExecutionResultConditions.instanceOf;
+import static org.junit.platform.testkit.engine.TestExecutionResultConditions.message;
+import static org.junit.platform.testkit.engine.TestExecutionResultConditions.throwable;
+import static org.junitpioneer.testkit.assertion.PioneerAssert.assertThat;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.UnknownHostException;
+import java.nio.file.FileAlreadyExistsException;
+import java.util.List;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junitpioneer.testkit.ExecutionResults;
+import org.junitpioneer.testkit.PioneerTestKit;
+
+@DisplayName("Resources extension")
+class ResourcesTests {
```

Comment:
I think we need to test the contract of the general-resource-portion of our extension(s). I will leave some comments below where I think that should be tested.

Reply:
Sounds good. I'll resolve this particular thread then and tackle the specific ones below.